File: sdk/nodejs/src/model/system/audit-log.ts



export const AuditLogResource = {
    resource: "AuditLog",
    namespace: "system",
};

// Sub Types

// Resource Type
export interface AuditLog {
    id: string;
version: number;
namespace: string;
resource: string;
recordId: string;
time: string;
username: string;
operation: 'CREATE' | 'UPDATE' | 'DELETE';
properties?: object;
annotations?: object;

}
// Resource and Property Names
export const AuditLogName = "AuditLog";

export const AuditLogIdName = "Id";

export const AuditLogVersionName = "Version";

export const AuditLogNamespaceName = "Namespace";

export const AuditLogResourceName = "Resource";

export const AuditLogRecordIdName = "RecordId";

export const AuditLogTimeName = "Time";

export const AuditLogUsernameName = "Username";

export const AuditLogOperationName = "Operation";

export const AuditLogPropertiesName = "Properties";

export const AuditLogAnnotationsName = "Annotations";

File: sdk/python/apibrew/model/audit_log.py
from datetime import datetime
from typing import List, Optional
from enum import Enum
from apibrew.entity import Entity, EntityInfo
from __future__ import annotations





class Operation(Enum):
    CREATE = "CREATE"
    UPDATE = "UPDATE"
    DELETE = "DELETE"



class AuditLog:
    id: str
    version: int
    namespace: str
    resource: str
    recordId: str
    time: datetime
    username: str
    operation: Operation
    properties: dict
    annotations: dict[str, str]

    @staticmethod
    def entity_info():
        return EntityInfo("system", "AuditLog", "system-auditlog")




